### PR TITLE
Report Violation Per Trade

### PR DESCRIPTION
**Pull request by akshatmittal**
Resolves https://github.com/code-423n4/2023-06-reserve-findings/issues/47

### CODE REVIEW

File: contracts/interfaces/IBroker.sol [hunk 1]
```diff
@@ -38,7 +38,7 @@ interface IBroker is IComponent {
     event DutchTradeImplementationSet(ITrade indexed oldVal, ITrade indexed newVal);
     event BatchAuctionLengthSet(uint48 indexed oldVal, uint48 indexed newVal);
     event DutchAuctionLengthSet(uint48 indexed oldVal, uint48 indexed newVal);
-    event DisabledSet(bool indexed prevVal, bool indexed newVal);
+    event DisabledSet(TradeKind tradeKind, bool indexed prevVal, bool indexed newVal);
```
Comment (tbrent) [inline contracts/interfaces/IBroker.sol:41]: Should be indexed I think

File: contracts/p0/Broker.sol [hunk 2]
```diff
@@ -68,7 +71,7 @@ contract BrokerP0 is ComponentP0, IBroker {
         TradeRequest memory req,
         TradePrices memory prices
     ) external returns (ITrade) {
-        require(!disabled, "broker disabled");
+        require(!tradeKindDisabled[kind], "broker disabled");
```
Comment (pmckelvy1) [inline contracts/p0/Broker.sol:74]: "broker disabled for that trade kind"

**Review by pmckelvy1 — CHANGES_REQUESTED** [review]

File: test/Broker.test.ts [hunk 3]
```diff
@@ -353,38 +354,43 @@ describe(`BrokerP${IMPLEMENTATION} contract #fast`, () => {
...
-      expect(await broker.disabled()).to.equal(false)
+      expect(await broker.tradeKindDisabled(TradeKind.DUTCH_AUCTION)).to.equal(false)
     })
   })
 
   describe('Trade Management', () => {
     it('Should not allow to open trade if Disabled', async () => {
       // Disable Broker
-      await expect(broker.connect(owner).setDisabled(true))
+      await expect(broker.connect(owner).setDisabled(TradeKind.DUTCH_AUCTION, true))
```
Comment (tbrent) [inline test/Broker.test.ts:388]: I'm going to tweak these tests to test each kind separately

**Comment by pmckelvy1** [comment]
github seems to be having an issue showing the proper diffs but they pull correctly so im going to merge

lgtm

**Comment by tbrent** [comment]
Oh I wasn't planning on merging this if we accepted #869, since they're basically mutually exclusive changes. I should be able to merge it though, just will take some work...

**Comment by akshatmittal** [comment]
Yeah I was also going to close this in favor of #869. Seems ok, should be fine.

**Pull request merged**